### DISCUSSION TITLE
L language中的language如何通俗地理解？

### DISCUSSION THREAD

**Discussion by passtheexam** [zkp-co-learning/zkp-co-learn#128] [Q&A（提问题在隔壁~）]
文献中经常会有L language、NP language等表述，请问这个“language”如果理解比较好呢？
是“问题”、“关系”的意思吗？

**Comment by Demian101** [comment]
可以具体举一个带有 Scene 的栗子吗

**Reply by passtheexam** [reply]
来源：https://crypto-notes-erhant.vercel.app/zklearning/introduction-and-history.html

![image](https://github.com/zkp-co-learning/zkp-co-learn/assets/95533028/144cde3d-c6dd-44c1-8815-74bc13cfc1a1)

![image](https://github.com/zkp-co-learning/zkp-co-learn/assets/95533028/0fdaf670-e33e-4cf4-9417-8877263bc764)

![image](https://github.com/zkp-co-learning/zkp-co-learn/assets/95533028/3bf2e500-8b75-429a-84f1-bf59fdda8ab0)

**Comment by luckyyang** [comment]
可以看这里的定义: https://youtu.be/i2nxdrMsEEk
A language is a set of binary string x that satisfy some properties.
![2023-08-13 at 12 23](https://github.com/zkp-co-learning/zkp-co-learn/assets/1963646/7111a310-17a4-4e6b-8ff8-eb2effb0d87e)

**Comment by timimm** [comment]
你可以这么理解：语言就是一类陈述（statement）的集合，陈述在计算机里面就表现为一段二进制字符串。重点在于如何对语言进行分类，比如：P语言，NP语言。对语言的分类指标在于它所包含的陈述的正确性的验证难度。
比如，给定陈述“数1000000000000....1是两个大质数的乘积”，验证这个陈述的正确性是困难的，因为目前不确定是否有多项式时间算法可以验证这个陈述，也就是NP的，那么我们说“数N是两个大质数的乘积”这一类陈述所构成的语言是NP语言。同理可以给出P语言的定义。可以看出NP语言的建立的基础是数学难题，这也是ZKP所需要的语言，因此ZKP中要求给出的陈述要属于一个NP语言，否则无法构建ZKP协议，像图同构、三染色等都是典型的NP语言。
以上只是本菜鸟的个人想法，语言不严谨，还请密码学大佬轻喷）。
